### DISCUSSION TITLE
Dual Funding Flow

### DISCUSSION THREAD

**Discussion by n1bor** [ACINQ/eclair#2377] [Developers]
Saw:
https://github.com/ACINQ/eclair/commit/0310bf5dc442d810636059fc371a970e6688f2f1

Is there a proposed API for this yet? Would be keen to test when there is.

Thanks

**Comment by t-bast** [comment]
Not yet, there are three more PRs that need to be merged to have a complete dual funding flow available.

This will initially just reuse the existing `open` API and opportunistically do dual funding if both peers support it. There will be a single funder in practice though, the longer term goal is to allow plugins to interface and choose when to add funds as a non-initiator. I will work on liquidity ads right after dual funding, because that's the only case I can think of where we'll have channels funded by both participants.

Do you have a specific usage in mind for this?
